dns_entry_checker/dns_entry_checker.py

```diff
@@ -0,0 +1,223 @@
+
+
+def check_output_files(
+    forward_mismatch_filepath,
+    backward_mismatch_filepath,
+    backward_missing_filepath,
+    gap_missing_filepath,
+):
+    """
+    Function to check if output files already exist
+        :param forward_mismatch_filepath: The filepath of the forward mismatch output file (String)
+        :param backward_mismatch_filepath: The filepath of the backward mismatch output file (String)
+        :param backward_missing_filepath: The filepath of the backward missing output file (String)
+        :param gap_missing_filepath: The filepath of the gap missing output file (String)
+    """
+    if os.path.exists(forward_mismatch_filepath):
+        raise RuntimeError(f"{forward_mismatch_filepath} already exists")
+    if os.path.exists(backward_mismatch_filepath):
+        raise RuntimeError(f"{backward_mismatch_filepath} already exists")
+    if os.path.exists(backward_missing_filepath):
+        raise RuntimeError(f"{backward_missing_filepath} already exists")
+    if os.path.exists(gap_missing_filepath):
+        raise RuntimeError(f"{gap_missing_filepath} already exists")
+
+
+def dns_entry_checker():
+    # Get arguments passed to the script
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawDescriptionHelpFormatter
+    )
+    parser.add_argument(
+        "-u", "--user", metavar="USER", help="FedID of the user", required=True
+    )
+    parser.add_argument(
+        "-p",
+        "--password",
+        metavar="PASSWORD",
+        help="Password of the user",
+        required=True,
+    )
+    parser.add_argument(
+        "-i",
+        "--ip",
+        metavar="IP",
+        help="IP of the machine to SSH to",
+        required=True,
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        metavar="OUTPUT",
+        help="Directory to create the output files in",
+    )
+
+    # Define the variables with the script arguments
+    args = parser.parse_args()
+    user = args.user
+    password = args.password
+    ip = args.ip
+    output = args.output
+
+    # Create an SSH client with the credentials given
+    client = create_client(ip, user, password)
+
+    # Run a command on the VM to get all IP addresses with their domain names
+    command_outputs = ssh_command(
+        client,
+        "dig @ns1.rl.ac.uk stfc.ac.uk  axfr | "
+        "grep -i nubes.stfc.ac.uk | "
+        "grep -v CNAME | "
+        "grep '172.16.'",
+    )
+
+    # Define the filepath for the output to be saved to
+    forward_mismatch_filepath = os.path.join(
+        output or "output", "forward_mismatch_list.txt"
+    )
+    backward_mismatch_filepath = os.path.join(
+        output or "output", "backward_mismatch_list.txt"
+    )
+    backward_missing_filepath = os.path.join(
+        output or "output", "backward_missing_list.txt"
+    )
+    gap_missing_filepath = os.path.join(output or "output", "gap_missing_list.txt")
+    check_output_files(
```

Comment:
you can rewrite check_output_files function to be more generic and call it multiple times for each file
something like 

```
for filepath in [forward_mismatch_filepath, backward_mismatch_filepath, backward_missing_filepath, gap_missing_filepath]:
    check_file(filepath)
```